httpsig.go

```diff
@@ -69,6 +69,7 @@ func NewSignTransport(transport http.RoundTripper, opts ...signOption) http.Roun
 
 			if n != 0 {
 				r.Body = io.NopCloser(bytes.NewReader(b.Bytes()))
+				nr.Body = io.NopCloser(bytes.NewReader(b.Bytes()))
```

Comment:
nice find; thank you! with this, i think we can remove the setting of `r.Body`. it looks like I put it on the wrong request.

Reply:
I'll check if r.Body is needed, I have to dig the test code I had written to highlight the issue but worked on so many things that I'm not sure where it is anymore :-)